halibot/message.py

```diff
@@ -59,6 +73,10 @@ def __setattr__(self, key, value):
 			object.__setattr__(self.context, 'agent', ls[0])
 			object.__setattr__(self.context, 'whom', '/'.join(ls[1:]))
 
+		if key == 'author':
+			object.__setattr__(self, key, self.Author(value))
+			return
```

Comment:
We should only be doing this is if the value is a string object. It is reasonable that an Author object is being assigned to the author field.